src/xERC20/contracts/XUltraLRT.sol

```diff
@@ -133,14 +133,52 @@ contract XUltraLRT is
         Message memory message = abi.decode(_message, (Message));
         // handle message
         if (message.msgType == MSG_TYPE.MINT) {
-            // TODO: introduce limits use burn with caller
-            _mint(message.sender, message.amount);
+            _handleCrossChainMint(message.sender, message.amount);
         } else if (message.msgType == MSG_TYPE.BURN) {
             _burn(message.sender, message.amount);
         } else if (message.msgType == MSG_TYPE.PRICE_UPDATE) {
             _updatePrice(message.price, message.timestamp);
         }
     }
+    /**
+     * @notice Handle cross chain mint
+     * @param _sender The sender address
+     * @param _amount The amount of token to mint
+     */
+
+    function _handleCrossChainMint(address _sender, uint256 _amount) internal {
+        // mint shares for the user
+        _mint(_sender, _amount);
+        // @dev need to mint first otherwise it will revert if supply is less that new limit
+        // increase cross chain transfer limit
+        // destination transfer limit will be decreased
+        _increaseCrossChainTransferLimit(_amount);
```

Comment:
handle L1 case, it should not have transfer limit

Reply:
we made it generic, each mint will increase limit.